### ISSUE TITLE
Configureable constant missing from ABI when only used in the storage block

### ISSUE THREAD

**Issue by K1-R1** [FuelLabs/sway#5242]
An example of this from a contract with a configureable constant `DEPLOYER` that is only used in the storage block:
```
configurable {
    DEPLOYER: b256 = 0x09c0b2d1a486c439a87bcba6b46a7a1a23f3897cc83a94521a96da5c23bc58db,
}

storage {
    deployer: Ownership = Ownership::initialized(Identity::Address(Address::from(DEPLOYER))),
...
```
It appears that this results in the configureable not being included in the ABI, as seen in more detail in the forum post below.
https://forum.fuel.network/t/no-configurable-field-in-abi-pyth-oracle-contract/3397

**Comment by chlenc** [comment]
are there any updates, frens?

**Comment by xunilrj** [comment]
The issue here is that configurables not being used are stripped from the ABI. In this case this configurable is only being used as the initializer of a storage, which is evaluated at compile time.

The same thing would happen, for example, in:

```sway
const a = DEPLOYER;
log(a);
```

In the end this configurable is not configuring anything, and because of that, my suggestion is to turn this into a warning: "this configurable is not being used and it is not available in the ABI", or something like that.

**Comment by xunilrj** [comment]
I created another issue to improve the diagnostic in this case. Can I close this as this is not a bug?

**Comment by K1-R1** [comment]
Closed in favour of #5261
